src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/X509Certificates/CertificatePal.Windows.PrivateKey.cs

```diff
@@ -682,39 +282,15 @@ private static bool TryGuessDsaKeySpec(CspParameters cspParameters, out int keyS
             return pal;
         }
 
-        private CertificatePal CopyWithEphemeralKey(CngKey cngKey)
+        private T? GetPrivateKey<T>(Func<CspParameters, T> createCsp, Func<CngKey, T?> createCng)
+            where T : class, IDisposable
         {
-            Debug.Assert(string.IsNullOrEmpty(cngKey.KeyName));
-
-            // Handle makes a copy of the handle.  This is required given that CertSetCertificateContextProperty accepts a SafeHandle
-            // and transfers ownership of the handle to the certificate.  We can't transfer that ownership out of the cngKey, as it's
-            // owned by the caller, so we make a copy (using Handle rather than HandleNoDuplicate).
-            using (SafeNCryptKeyHandle handle = cngKey.Handle)
-            {
-                // Make a new pal from bytes.
-                CertificatePal pal = (CertificatePal)FromBlob(RawData, SafePasswordHandle.InvalidHandle, X509KeyStorageFlags.PersistKeySet);
-                try
-                {
-                    if (!Interop.Crypt32.CertSetCertificateContextProperty(
-                        pal._certContext,
-                        Interop.Crypt32.CertContextPropId.CERT_NCRYPT_KEY_HANDLE_PROP_ID,
-                        Interop.Crypt32.CertSetPropertyFlags.CERT_SET_PROPERTY_INHIBIT_PERSIST_FLAG,
-                        handle))
-                    {
-                        throw Marshal.GetLastPInvokeError().ToCryptographicException();
-                    }
-
-                    // The value was transferred to the certificate.
-                    handle.SetHandleAsInvalid();
-
-                    return pal;
-                }
-                catch
-                {
-                    pal.Dispose();
-                    throw;
-                }
-            }
+            return CertificateHelpers.GetPrivateKey<T>(this, createCsp, createCng);
         }
+
+
+        private CertificatePal? CopyWithPersistedCngKey(CngKey cngKey) => CertificateHelpers.CopyWithPersistedCngKey(this, cngKey);
```

Comment:
Nit

```suggestion
        }

        private CertificatePal? CopyWithPersistedCngKey(CngKey cngKey) => CertificateHelpers.CopyWithPersistedCngKey(this, cngKey);
```

Reply:
Created https://github.com/dotnet/runtime/issues/118066 for follow-up